lib/workers/repository/update/pr/changelog/index.ts

```diff
@@ -44,6 +68,8 @@ export async function getChangeLogJSON(
         break;
     }
 
+    config.sourceUrl = srcUrlContainer.restore();
```

Comment:
Why is it needed to restore it?

Reply:
> Note: sourceUrl is restored to its original value after changelog retrieval, this is due the fact that it is part of the repository cache. and if im not mistaken changelogs has no meaning in the context of repository cache but the original datasource fetched sourceUrl does.

basically i want it to be used as part of the repo cache for the merge confidence. in this context the user set URL is only for changelogs retravel but the `sourceUrl` set by the `datasource` is what we expect for a given `dep` and what we already have.

Reply:
I forget, how are we using sourceUrl in merge confidence?

Reply:
it is part of `getUpdatesFromCache`, each array element has a `sourceUrl` prop

Reply:
don't know how this data is processed though

Reply:
if cache is updated, then it is used to `updateRepoBranches` in the db